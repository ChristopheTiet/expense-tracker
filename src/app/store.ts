import { configureStore } from "@reduxjs/toolkit";
import expensesReducer from "../ducks/expenses-slice";
import filterReducer from "../ducks/filter-slice";

export const store = configureStore({
  reducer: {
    expenses: expensesReducer,
    filter: filterReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
