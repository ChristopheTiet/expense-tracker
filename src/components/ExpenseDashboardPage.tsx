import ExpenseList from "./ExpenseList";
import { useDispatch } from "react-redux";
import { addExpense } from "../ducks/expenses-slice";
import ExpenseListFilter from "./ExpenseListFilter";

export default function ExpenseDashboardPage() {
  const dispatch = useDispatch();
  dispatch(
    addExpense({
      description: "test1",
      note: "note",
      amount: 50,
      createdAt: 100,
    })
  );
  dispatch(
    addExpense({
      description: "test2",
      note: "note2",
      amount: 100,
      createdAt: 50,
    })
  );
  dispatch(
    addExpense({
      description: "test3",
      note: "note3",
      amount: 150,
      createdAt: 0,
    })
  );
  return (
    <div className="p-4">
      <ExpenseListFilter />
      <ExpenseList />
    </div>
  );
}
