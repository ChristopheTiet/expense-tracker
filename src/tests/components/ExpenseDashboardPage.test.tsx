import renderer from "react-test-renderer";
import ExpenseDashboardPage from "../../components/ExpenseDashboardPage";
import { BrowserRouter } from "react-router-dom";
import { store } from "../../app/store";
import { Provider } from "react-redux";

test("should render ExpenseDashBoard correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <ExpenseDashboardPage />
        </Provider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
