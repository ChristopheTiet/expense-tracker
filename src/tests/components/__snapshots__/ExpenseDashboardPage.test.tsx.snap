// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseDashBoard correctly 1`] = `
<div
  className="p-4"
>
  <div
    className="flex space-x-3"
  >
    <input
      className="text-base font-normal border-black border-2 rounded pl-5"
      onChange={[Function]}
      placeholder="filter"
      type="text"
      value=""
    />
    <select
      aria-label="order by amount or date"
      className=" text-base font-normal text-gray-700 bg-white bg-clip-padding bg-no-repeat border-2 border-solid border-gray-300 rounded transition focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none"
      onChange={[Function]}
      value="date"
    >
      <option
        value="amount"
      >
        amount
      </option>
      <option
        value="date"
      >
        date
      </option>
    </select>
    <div>
      From:
       
      <input
        className=" text-base font-normal text-gray-700 bg-white bg-clip-padding bg-no-repeat border-2 border-solid border-gray-300 rounded transition focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none"
        onChange={[Function]}
        type="date"
      />
    </div>
    <div>
      To:
       
      <input
        className=" text-base font-normal text-gray-700 bg-white bg-clip-padding bg-no-repeat border-2 border-solid border-gray-300 rounded transition focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none"
        onChange={[Function]}
        type="date"
      />
    </div>
  </div>
  <div
    className="space-y-2 py-4"
  >
    <p>
      No expenses
    </p>
  </div>
</div>
`;
